src/tests/async.test.ts

```diff
@@ -256,13 +257,7 @@ describe("async module", () => {
         await _.sleep(300)
         numInProgress--;
       })
-      assert.deepEqual(tracking, [
-        1, 2, 3, 
-        3, 3, 3, 
-        3, 3, 3, 
-        3, 3, 3, 
-        3, 3
-      ])
+      assert.deepEqual(Math.max(...tracking), 3)
```

Comment:
What's the reason for changing this test? Only asserting the max could lead to false positives in this case.

Reply:
Because this test produces the different result with fake timers. The limit argument defines how many functions should be allowed to run at once, right?

For example, https://replit.com/@kmisha/radash-parallel-example?v=1. The number for running task not always should be equal to the limit.

Reply:
Ya, that's correct about the limit argument 👍 

Maybe we have to redesign this test completely, I've pulled it locally and started playing with it. I see what you're saying, the fake timers make the tracking `[1, 2, 3, 1, 2, 3, 1, 2, 3]`.

Somehow, we still need to be able to assert that for a long running task (I used 300ms to be long running) parallel still only executes the limit as a max.

I've been reading up on the fake timers, maybe this test/assertion still does that, I'm just not confident it does. I could be wrong tho?

Reply:
I'm going to approve and merge, I'll try to figure out this test in another PR.